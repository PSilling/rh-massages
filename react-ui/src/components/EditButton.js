// react imports
import React, { Component } from 'react';

/**
 * Custom edit button component.
 */
class EditButton extends Component {

  render() {
    return(
      <span>
        <button style={{ 'color': '#000' }}  type="button" className="btn btn-link" onClick={this.props.onEdit}>
          <span className="glyphicon glyphicon-pencil"></span>
        </button>
      </span>
    );
  }
}

export default EditButton
