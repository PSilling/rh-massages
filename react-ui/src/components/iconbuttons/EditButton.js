// react imports
import React from "react";
import PropTypes from "prop-types";

// util imports
import _t from "../../util/Translations";

/**
 * Icon only button used for element editing.
 */
const EditButton = function EditButton(props) {
  return (
    <span>
      <button
        style={{ color: "#000" }}
        type="button"
        className="btn btn-link"
        onClick={props.onEdit}
        title={_t.translate("Edit")}
      >
        <span className="glyphicon glyphicon-pencil" />
      </button>
    </span>
  );
};

EditButton.propTypes = {
  /** function to be called on button click */
  onEdit: PropTypes.func.isRequired
};

export default EditButton;
