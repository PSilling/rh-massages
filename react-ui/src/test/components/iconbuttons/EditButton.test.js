// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import EditButton from '../../../components/iconbuttons/EditButton';

test('renders content with correct props', () => {
  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <EditButton onEdit={testFunction} />
        ),
        testInstance = testRenderer.root;

  let button = testInstance.findByType('button'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-pencil" }),
      treeJSON = testRenderer.toJSON();

  expect(testFunction).not.toHaveBeenCalled();

  button.props.onClick();

  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(button.props.onClick).toBe(testFunction);
  expect(treeJSON).toMatchSnapshot();
});
