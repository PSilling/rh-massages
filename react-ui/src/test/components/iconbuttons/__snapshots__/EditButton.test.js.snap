// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<span>
  <button
    className="btn btn-link"
    onClick={[Function]}
    style={
      Object {
        "color": "#000",
      }
    }
    title="Upravit"
    type="button"
  >
    <span
      className="glyphicon glyphicon-pencil"
    />
  </button>
</span>
`;
